File: src/automatica.core/Automatica.Core.Model/Models/User/User2Role.cs
﻿using System;

namespace Automatica.Core.Model.Models.User
{
    public class User2Role : TypedObject
    {
        public Guid This2User { get; set; }
        public Guid This2Role { get; set; }


        public User This2UserNavigation { get; set; }
        public Role This2RoleNavigation { get; set; }
    }
}

File: src/automatica.core/Automatica.WebNew/src/app/base/model/user/user2role.ts
import { BaseModel, Model, JsonFieldInfo, JsonProperty } from "../base-model";
import { Role } from "./role";

@Model()
export class User2Role extends BaseModel {

    @JsonProperty()
    This2Role: string;

    @JsonProperty()
    This2User: string;

    @JsonProperty()
    This2RoleNavigation: Role;

    public typeInfo(): string {
        return "User2Role";
    }

    protected getJsonProperty(): Map<string, JsonFieldInfo> {
        return new Map<string, JsonFieldInfo>();
    }
}
